mod/ckeditor/views/default/elgg/ckeditor/config.js

```diff
@@ -1,23 +1,10 @@
+/**
+ * Do not require this module. Use elgg/ckeditor#config value instead
+ * @deprecated 2.2
+ */
 define(function(require) {
-	require('elgg/init');
 	var elgg = require('elgg');
-	var $ = require('jquery');
-
-	return elgg.trigger_hook('config', 'ckeditor', null, {
-		toolbar: [['Bold', 'Italic', 'Underline', 'RemoveFormat'], ['Strike', 'NumberedList', 'BulletedList', 'Undo', 'Redo', 'Link', 'Unlink', 'Image', 'Blockquote', 'Paste', 'PasteFromWord', 'Maximize']],
-		removeButtons: 'Subscript,Superscript', // To have Underline back
-		allowedContent: true,
-		baseHref: elgg.get_site_url(),
-		removePlugins: 'liststyle,contextmenu,tabletools,resize',
-		extraPlugins: 'blockimagepaste',
-		defaultLanguage: 'en',
-		language: elgg.get_language(),
-		skin: 'moono',
-		uiColor: '#EEEEEE',
-		contentsCss: elgg.get_simplecache_url('elgg/wysiwyg.css'),
-		disableNativeSpellChecker: false,
-		disableNativeTableHandles: false,
-		removeDialogTabs: 'image:advanced;image:Link;link:advanced;link:target',
-		autoGrow_maxHeight: $(window).height() - 100
-	});
+	var elggCKEditor = require('elgg/ckeditor');
```

Comment:
There is a risk here that someone replaced the `elgg/ckeditor` module and calls `elgg/ckeditor/config`, so we will end up in an infinite loop.